math/src/field/fields/montgomery_backed_prime_fields.rs

```diff
@@ -385,6 +385,36 @@ where
     }
 }
 
+impl<M, const NUM_LIMBS: usize> HasDefaultTranscript for MontgomeryBackendPrimeField<M, NUM_LIMBS>
+where
+    M: IsModulus<UnsignedInteger<NUM_LIMBS>> + Clone + Debug,
+{
+    /// # Panics
+    ///
+    /// This function will panic if NUM_LIMBS is greater than 4.
+    fn get_random_field_element_from_seed(
+        seed: [u8; 32],
+    ) -> Option<FieldElement<MontgomeryBackendPrimeField<M, NUM_LIMBS>>> {
+        let mut sample = UnsignedInteger::from_bytes_be(&seed).unwrap();
+
+        let first_non_zero_limb_index = M::MODULUS.limbs.iter().position(|&x| x != 0).unwrap_or(0);
+
+        for i in 0..first_non_zero_limb_index {
+            sample.limbs[i] = 0;
+        }
+
+        let mask = u64::MAX >> M::MODULUS.limbs[first_non_zero_limb_index].leading_zeros();
+        sample.limbs[first_non_zero_limb_index] &= mask;
+
+        if sample > M::MODULUS {
```

Comment:
As elsewhere, `sample == MODULUS` should be rejected.

```suggestion
        if sample >= M::MODULUS {
```